internal/diagnostics/types.go

```diff
@@ -0,0 +1,62 @@
+package diagnostics
+
+import (
+	"github.com/kong/go-database-reconciler/pkg/file"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+)
+
+// DumpMeta annotates a config dump.
+type DumpMeta struct {
+	// Failed indicates the dump was not accepted by the Kong admin API.
+	Failed bool
+	// Fallback indicates that the dump is a fallback configuration attempted after a failed config update.
+	Fallback bool
+	// AffectedObjects are objects excluded from the fallback configuration.
+	AffectedObjects []AffectedObject
+	// Hash is the configuration hash.
+	Hash string
+}
+
+// ConfigDump contains a config dump and a flag indicating that the config was not successfully applid.
+type ConfigDump struct {
+	// Config is the configuration KIC applied or attempted to apply.
+	Config file.Content
+	// Meta contains information about the status and context of the configuration dump.
+	Meta DumpMeta
+	// RawResponseBody is the raw Kong Admin API response body from a config apply. It is only available in DB-less mode.
+	RawResponseBody []byte
+}
+
+type configDumpResponse struct {
+	ConfigHash string       `json:"hash"`
+	Config     file.Content `json:"config"`
+}
+
+type problemObjectsResponse struct {
+	ConfigHash string           `json:"hash"`
+	Objects    []AffectedObject `json:"objects"`
```

Comment:
We want to add `excludedObjects` and `backfilledObjects` as well so that might make sense to make this name more precise, WDYT?
```suggestion
	BrokenObjects    []AffectedObject `json:"brokenObjects,omitempty"`
```